sdk/core/azure-core/src/main/java/com/azure/core/http/policy/AzureSasCredentialPolicy.java

```diff
@@ -74,4 +74,13 @@ public AzureSasCredentialPolicy(AzureSasCredential credential, boolean requireHt
 
         return next.process();
     }
+
+    /**
+     * Returns the signature stored by this credential's policy.
+     *
+     * @return The signature stored by this credential's policy
+     */
+    public String getSignature() {
+        return credential.getSignature();
+    }
```

Comment:
When we extract the signature as `String` to use in other client then the user loses ability to use `AzureSasCredential.update()` API to rotate the SAS credential, so this might potentially introduce regressions for such scenarios.
If I understand related bug the signature is getting lost somewhere when new clients are produced internally in the SDK. In principle those should inherit the pipeline from originating client. What's the mechanism the signature is getting lost in the `DataLakeFileClient rename(String, String)`, i.e. is `AzureSasCredentialPolicy` getting lost when constructing return type ?

Reply:
Sorry, I got the second part wrong.

Reply:
Looks like the bare minimum here would be to enrich javadocs to indicate that user should always call this method to obtain latest value of the signature, i.e. that the returned value must not be cached.